src/search.rs

```diff
@@ -0,0 +1,182 @@
+            let mut mut_salt = [0u8; 32];
+            take_lock(&self.rng).deref_mut().fill_bytes(&mut mut_salt);
+            mut_salt
+        };
+
+        let encrypted_salt = self
+            .document_encrypt_unmanaged(
+                &salt,
+                &DocumentEncryptOpts::with_explicit_grants(
+                    None,
+                    None,
+                    false,
+                    vec![group_id.into()],
+                ),
+            )
+            .await?;
+        let search = BlindIndexSearch::new(salt);
+
+        Ok(BlindIndexCreateResult {
+            encrypted_salt: encrypted_salt.try_into()?,
+            sdk: search,
+        })
+    }
+}
+#[derive(Debug, PartialEq, Clone, Hash, Eq)]
+pub struct BlindIndexSearch {
+    decrypted_salt: [u8; 32],
+}
+
+impl TryFrom<&[u8]> for BlindIndexSearch {
+    type Error = IronOxideErr;
+    fn try_from(bytes: &[u8]) -> Result<BlindIndexSearch> {
+        let decrypted_len = bytes.len();
+        if decrypted_len != REQUIRED_LEN {
+            std::result::Result::Err(IronOxideErr::WrongSizeError(
+                Some(decrypted_len),
+                Some(REQUIRED_LEN),
+            ))
+        } else {
+            let mut a = [0u8; 32];
+            a.copy_from_slice(&bytes[0..32]);
+            Ok(BlindIndexSearch::new(a))
+        }
+    }
+}
+
+impl TryFrom<DocumentEncryptUnmanagedResult> for EncryptedBlindIndexSalt {
+    type Error = IronOxideErr;
+    fn try_from(r: DocumentEncryptUnmanagedResult) -> Result<EncryptedBlindIndexSalt> {
+        match r.access_errs().get(0) {
+            None => Ok(EncryptedBlindIndexSalt {
+                encrypted_deks: r.encrypted_deks().to_vec(),
+                encrypted_salt_bytes: r.encrypted_data().to_vec(),
+            }),
+            Some(err) => Err(IronOxideErr::UserOrGroupDoesNotExist(
+                err.user_or_group.clone(),
+            )),
+        }
+    }
+}
+
+impl BlindIndexSearch {
+    fn new(decrypted_salt: [u8; 32]) -> BlindIndexSearch {
+        BlindIndexSearch { decrypted_salt }
+    }
+
+    ///Generate the list of tokens to use to find entries that match the search query, given the specified partition_id.
+    /// query - The string you want to tokenize and hash
+    /// partition_id - An extra string you want to include in every hash, this allows 2 queries with different partition_ids to produce a different set of tokens for the same query
+    pub fn tokenize_query(&self, query: &str, partition_id: Option<&str>) -> HashSet<u32> {
+        generate_hashes_for_string(query, partition_id, &self.decrypted_salt[..])
+    }
+
+    ///Generate the list of tokens to use to find entries that match the search query, given the specified partition_id.
+    /// query - The string you want to tokenize and hash
+    /// partition_id - An extra string you want to include in every hash, this allows 2 queries with different partition_ids to produce a different set of tokens for the same data
```

Comment:
This comment looks like it was just copy/pasted from `tokenize_query` where I assume it serves a different purpose even though the arguments are the same type.

Reply:
That's because they currently do the same thing. I'll change it when it gets updated to do something else.

Reply:
That time has come

Reply:
I changed the comment. Let me know if you think it could be improved.